app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -334,7 +334,20 @@ def template_from_vmdb(deployment)
 
       def download_template(uri)
         require 'open-uri'
-        open(uri) { |f| f.read }
+
+        if @config.proxy.blank?
+          open(uri) { |f| f.read }
+        else
+          proxy = Addressable::URI.parse(@config.proxy)
+          proxy_url = "#{proxy.scheme}://#{proxy.host}:#{proxy.port}"
+
+          if proxy.user.blank?
+            open(uri, :proxy => proxy_url) { |f| f.read }
+          else
+            auth_array = [proxy_url, proxy.user, proxy.password]
+            open(uri.to_s, :proxy_http_basic_authentication => auth_array) { |f| f.read }
```

Comment:
@djberg96 Were there tests for this conditional logic or was that coming in the followup PR?  If not, please add them.  Thanks!